src/main/java/net/snowflake/ingest/streaming/internal/OpenChannelRequestInternal.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright (c) 2024 Snowflake Computing Inc. All rights reserved.
+ */
+
+package net.snowflake.ingest.streaming.internal;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import net.snowflake.ingest.streaming.OpenChannelRequest;
+import net.snowflake.ingest.utils.Constants;
+import net.snowflake.ingest.utils.Utils;
+
+/** Class used to serialize the {@link OpenChannelRequest} */
+class OpenChannelRequestInternal implements StreamingIngestRequest {
+  @JsonProperty("request_id")
+  private String requestId;
+
+  @JsonProperty("role")
+  private String role;
+
+  @JsonProperty("channel")
+  private String channel;
+
+  @JsonProperty("table")
+  private String table;
+
+  @JsonProperty("database")
+  private String database;
+
+  @JsonProperty("schema")
+  private String schema;
+
+  @JsonProperty("write_mode")
+  private String writeMode;
+
+  @JsonProperty("is_iceberg")
+  private boolean isIceberg;
+
+  @JsonInclude(JsonInclude.Include.NON_NULL)
+  @JsonProperty("offset_token")
+  String offsetToken;
+
+  OpenChannelRequestInternal(
+      String requestId,
+      String role,
+      String database,
+      String schema,
+      String table,
+      String channel,
+      Constants.WriteMode writeMode,
+      String offsetToken,
+      boolean isIceberg) {
+    this.requestId = requestId;
+    this.role = role;
+    this.database = database;
+    this.schema = schema;
+    this.table = table;
+    this.channel = channel;
+    this.writeMode = writeMode.name();
+    this.isIceberg = isIceberg;
+    this.offsetToken = offsetToken;
+  }
+
+  String getRequestId() {
+    return requestId;
+  }
+
+  String getRole() {
+    return role;
+  }
+
+  String getChannel() {
+    return channel;
+  }
+
+  String getTable() {
+    return table;
+  }
+
+  String getDatabase() {
+    return database;
+  }
+
+  String getSchema() {
+    return schema;
+  }
+
+  String getWriteMode() {
+    return writeMode;
+  }
+
+  boolean getIsIceberg() {
+    return isIceberg;
+  }
+
+  String getOffsetToken() {
+    return offsetToken;
+  }
+
+  String getFullyQualifiedTableName() {
+    return Utils.getFullyQualifiedTableName(database, schema, table);
+  }
+
+  @Override
+  public String getStringForLogging() {
+    return String.format(
+        "OpenChannelRequestInternal(requestId=%s, role=%s, db=%s, schema=%s, table=%s, channel=%s,"
+            + " writeMode=%s, isIceberg=%s, offsetToken=%s)",
```

Comment:
not sure we log offset token in other places. if not, suggest to remove it as it can contain sensitive data

Reply:
Existed [here](https://github.com/snowflakedb/snowflake-ingest-java/blob/f5669d162603f621990538dd82eba5714a1b8faa/src/main/java/net/snowflake/ingest/streaming/internal/SnowflakeStreamingIngestClientInternal.java#L357) in the old logic. Should we remove it?